src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1343,25 +1365,46 @@ void uct_dc_mlx5_ep_pending_purge(uct_ep_h tl_ep, uct_pending_purge_callback_t c
     }
 }
 
-ucs_status_t uct_dc_mlx5_ep_check_fc(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
+ucs_status_t uct_dc_mlx5_ep_check_fc(uct_dc_mlx5_iface_t *iface,
+                                     uct_dc_mlx5_ep_t *ep)
 {
+    ucs_status_t ret_status = UCS_OK;
     ucs_status_t status;
 
     if (iface->super.super.config.fc_enabled) {
-        UCT_RC_CHECK_FC_WND(&ep->fc, ep->super.stats);
-        if (ep->fc.fc_wnd == iface->super.super.config.fc_hard_thresh) {
+        if (ucs_unlikely((ep->fc.fc_wnd ==
+                          iface->super.super.config.fc_hard_thresh) ||
+                         (ep->fc.fc_wnd <= 0))) {
+            if (ep->fc.fc_wnd <= 0) {
```

Comment:
can we avoid checking (ep->fc.fc_wnd <= 0) twice?

Reply:
done

Reply:
move this to function end, remove local var ret_status